client/dashboard/sites/site-reset-modal/index.tsx

```diff
@@ -0,0 +1,240 @@
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Button,
+	Modal,
+	TextControl,
+	ProgressBar,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect, useState, useCallback } from 'react';
+import {
+	siteResetContentSummaryQuery,
+	resetSiteMutation,
+	siteResetStatusQuery,
+} from '../../app/queries';
+import ExportNotice from '../export-content-notice';
+import ContentInfo from './content-info';
+import type { Site, SiteContent, SiteResetStatus } from '../../data/types';
+
+import './style.scss';
+
+interface ErrorResponse {
+	code: string;
+	message: string;
+}
+
+function ErrorContent( { message, onClose }: { message: string; onClose: () => void } ) {
+	return (
+		<VStack spacing={ 6 }>
+			<Text>{ message }</Text>
+			<HStack justify="flex-end">
+				<Button variant="primary" onClick={ onClose }>
+					{ __( 'Ok' ) }
+				</Button>
+			</HStack>
+		</VStack>
+	);
+}
+
+function InProgressContent( { progress }: { progress: number | undefined } ) {
+	const progressValue = progress ? progress * 100 : 10;
+	return (
+		<VStack spacing={ 4 }>
+			<ProgressBar className="reset-site-modal-progress-bar" value={ progressValue } />
+			<Text>{ __( "We're resetting your site. We'll email you once it's ready." ) }</Text>
+		</VStack>
+	);
+}
+
+function SiteResetContent( {
+	site,
+	siteContent,
+	siteDomain,
+	onSuccess,
+	onError,
+	onClose,
+}: {
+	site: Site;
+	siteContent: SiteContent;
+	siteDomain: string;
+	onSuccess: () => void;
+	onError: ( message: string ) => void;
+	onClose: () => void;
+} ) {
+	const { mutate } = useMutation< void, ErrorResponse >( resetSiteMutation( site.ID ) );
+	const [ isMutatePending, setIsMutatePending ] = useState( false );
```

Comment:
Why not use `isPending` from `useMutation` 🤔

Reply:
🤦 I was looking at [v4 documentation](http://tanstack.com/query/v4/docs/framework/react/reference/useMutation) which doesn't include `isPending`.